src/mono/wasm/runtime/driver.c

```diff
@@ -137,6 +143,22 @@ wasm_logger (const char *log_domain, const char *log_level, const char *message,
 	}, log_level, message, fatal, log_domain);
 }
 
+typedef uint32_t target_mword;
+typedef target_mword SgenDescriptor;
+typedef SgenDescriptor MonoGCDescriptor;
+MONO_API int   mono_gc_register_root (char *start, size_t size, MonoGCDescriptor descr, MonoGCRootSource source, void *key, const char *msg);
```

Comment:
This needs to be in mono, it can be done later.

Reply:
what would that look like? if it's simple I can do it in this PR

Reply:
I can do it after its merged.